src/solvers/dgsem_tree/subcell_limiters_2d.jl

```diff
@@ -13,22 +13,23 @@ function create_cache(limiter::Type{SubcellLimiterIDP}, equations::AbstractEquat
                                                                         nnodes(basis),
                                                                         bound_keys)
 
-    # Memory for bounds checking routine with `BoundsCheckCallback`.
+    # Threaded memory for bounds checking routine with `BoundsCheckCallback`.
     # The first entry of each vector contains the maximum deviation since the last export.
-    # The second one contains the total maximum deviation.
-    idp_bounds_delta = Dict{Symbol, Vector{real(basis)}}()
+    # In the second entry, the total maximum deviation is saved.
+    idp_bounds_delta_threaded = Dict{Symbol, Vector{Vector{real(basis)}}}()
     for key in bound_keys
-        idp_bounds_delta[key] = zeros(real(basis), 2)
+        idp_bounds_delta_threaded[key] = [zeros(real(basis), 2)
+                                          for _ in 1:Threads.nthreads()]
     end
 
-    return (; subcell_limiter_coefficients, idp_bounds_delta)
+    return (; subcell_limiter_coefficients, idp_bounds_delta_threaded)
 end
 
 function (limiter::SubcellLimiterIDP)(u::AbstractArray{<:Any, 4}, semi, dg::DGSEM, t,
                                       dt;
                                       kwargs...)
     @unpack alpha = limiter.cache.subcell_limiter_coefficients
-    alpha .= zero(eltype(alpha))
+    @trixi_timeit timer() "reset alpha" reset_du!(alpha, dg, semi.cache)
```

Comment:
Is it okay to reuse this function, even if the name can be confusing? I saw that `reset_du!` is also used for resetting some other variables. So, I used it as well.

Alternatively, I also can use the plane code of the function:
```suggestion
    @trixi_timeit timer() "reset alpha" @threaded for element in eachelement(dg, semi.cache)
        alpha[.., element] .= zero(eltype(alpha))
    end
```

Reply:
In TrixiParticles, we had the same discussion and settles with a generic function `set_zero!` for `du` and all other things that need to be set to zero before some kind of summation.

Reply:
Please create an issue for this but keep it here as it is, maybe with a comment
```suggestion
    # TODO: Do not abuse `reset_du!` but maybe implement a generic `set_zero!`
    @trixi_timeit timer() "reset alpha" reset_du!(alpha, dg, semi.cache)
```

Reply:
Done in [7d8313e](https://github.com/trixi-framework/Trixi.jl/pull/1736/commits/7d8313e302255759553534c1f2bebc7fc16a103e).

Reply:
Please also create an issue referencing this convo 🙏